docs/custom-planning-unit-grid-geometry.hld.md

```diff
@@ -0,0 +1,17 @@
+# Goal
+
+Users may sometimes create their own planning unit geometries, instead of
+relying on generated square or hexagonal grids, for a variety of conservation
+reasons, so they should be able to supply their own PU grid via a shapefile
+upload.
+
+# Key assumptions
+
+* once set, shape cannot be changed or removed from project
+* only `FeatureCollection` with `Polygon` features can be uploaded
+* custom planning unit grid geometry belongs to given project
+* custom planning unit grid geometry has a type of `irregular`
```

Comment:
`Irregular` is not for user-uploaded data, is for another kind of PU grid that will not follow a regular shape (i.e. River basins), and we will provide it in a future phase of the project. (Instead, please provide the options to the user)

Reply:
@aagm what should then be the type then? if this is a new type, what possible implications it has?

Reply:
The user should provide the type and the 3 we have should be kept (we should also expand it with from_shapefile:  [square, hexagon, irregular, from_shapefile]) and match the enum in the api planning_unit_grid_shape ( [square, hexagon, irregular, from_shapefile]), What do you think?

Reply:
added in https://github.com/Vizzuality/marxan-cloud/pull/526/commits/99dbd8f7e4de43afdabaf873430248257ef2880a